### DISCUSSION TITLE
PopoverContent  on Drawer is displayed under DrawerOverlays

### DISCUSSION THREAD

**Discussion by takeshiigarashi** [chakra-ui/chakra-ui#5974] [Q&A]
![screencapture-es8k3r-csb-app-2022-05-05-23_04_43](https://user-images.githubusercontent.com/27143411/166941850-0ca94758-ecc6-4ee7-8676-19a58349224e.png)

The problem is that PopoverContent  on Drawer is displayed under DrawerOverlays.
Popover is placed on Drawer.
Since the width of Popover content must be wider than Drawer's, 
PopoverContent is wrapped with Portal component.

https://codesandbox.io/s/popover-on-drawer-es8k3r?file=/src/index.js
https://es8k3r.csb.app/

How to fix?

z-index of PopoverContent does not work. z-index is not rendering to dom component.
Wrap PopoverContent by Box component having a specific css class name,
and write css file with "the-class-name > div { z-index: 1500; }".
The above work correctly. Popover has displayed over DrawerOverlays.
But z-index manipulation does not sound good.
Is there any good solution?

**Comment by nappalm** [comment]
any solution?

**Reply by bytesizelabsukltd** [reply]
Try passing portalZIndex={40} or any other number next to your ChakraProvider in your _app.tsx or similar. 

Like this

```
<ChakraProvider portalZIndex={40}>
   <App>
</ChakraProvider>
```

**Comment by tgmarinho** [comment]
I tried to debug this issue with my friend @Idaslon

But we cannot figure this issue, we prefered to create a drawer custom component:

```
// imports

interface Props {
  isOpen: boolean
}
/**
 * We created this component once we couldn't use the drawer from chakra-ui with the popover that is used in the share button
 *
 * @see https://github.com/chakra-ui/chakra-ui/discussions/5974
 *
 * @param {Props} { isOpen }
 * @return {*}
 */
export const BoxDrawer = ({ isOpen }: Props) => {
  return (
    <Slide direction={'bottom'} in={isOpen} style={{ zIndex: 'var(--chakra-zIndices-modal)' }}>
      <Box h={'calc(100dvh - 58px)'} bg={'blackAlpha.900'} p={6} mb={'58px'}>
        <Text fontSize={'24px'}>Title</Text>
        <Divider />
        <MyAboutContent />
      </Box>
    </Slide>
  )
}


```

clean solution for us.
